src/server.ts

```diff
@@ -59,14 +60,18 @@ export class Server {
       };
   }
 
-  private async generatePDF(url: string, res: Express.Response) {
+  private async generatePDF(url: string, async:boolean, res: Express.Response) {
     try {
       const browser = await puppeteer.launch({
           args: ['--no-sandbox']
         });
 
       const page = await browser.newPage();
-      await page.goto(url);
+      async
+          ?
+          await page.goto(url, {waitUntil: 'networkidle0'})
+          :
+          await page.goto(url);
```

Comment:
Maybe this is easier to read

```suggestion
      const params = async ? { waitUntil: 'networkidle0' } : {};
      
      await page.goto(url, params);
```